docs/trex_getdata.md

```diff
@@ -256,6 +347,36 @@ The following example returns the column names of the first logical table that i
 
 ```
 
+#### 2b. Get data from the logical table(s) for more than 10,000 rows
+
+Starting with Tableau 2022.4 and the Dashboard Extensions API library v1.10, if the data contains more than 10,000 rows, use the `getLogicalTableDataReaderAsync` method to get data from a data source. In this case, you create a `DataTableReader` to iterate through the pages of data for all rows in the worksheet. You process each `DataTable` sequentially. You can control the page size, using the optional `pageRowCount` parameter when you call `getLogicalTableDataReaderAsync`. The default page size is 10,000 rows.
+
+The basic steps are as follows:
+
+Using the `logicalTable.id`, call the `getLogicalTableDataReaderAsync` method to create the `DataTableReader`.
+Create a loop to retrieve each page of logical data, using `DataTableReader.pageCount` or `DataTableReader.totalRowCount` property of the summary `DataTableReader` to determine the number of pages to process. Use the `getPageAsync()` method to get the `DataTable` from each page. After you have retrieved all pages of the  data, call the `releaseAsync()` method to free up memory from the `DataTableReader`.
+
+```javascript
+
+// assumes this code is in an async method
+
+const pageRowCount = 1000;  // default is 10,000
+const dataSources = await worksheet.getDataSourcesAsync();
+const dataSource = dataSources.find(datasource => datasource.name === "Sample - Superstore");
+const logicalTables = await dataSource.getLogicalTablesAsync()
+const dataTableReader = await dataSource.getLogicalTableDataReaderAsync(logicalTables[0].id, pageRowCount);
+   
+const pageNumber = 0;
+   while (pageNumber < dataTableReader.pageCount) {
+      let currentPageDataTable = await dataTableReader.getPageAsync(pageNumber);
+      // process page ...
+      console.log(currentPageDataTable);
+      pageNumber++;
+    }
+// release resources
+await pageReader.releaseAsync();
```

Comment:
See comment above...
while to for
use finally for releaseAsync

Reply:
done